crates/sui-benchmark/src/bin/stress.rs

```diff
@@ -374,23 +390,72 @@ async fn main() -> Result<()> {
                 } => {
                     let shared_counter_ratio = 1.0
                         - (std::cmp::min(shared_counter_hotness_factor as u32, 100) as f32 / 100.0);
+                    let max_ops = target_qps * in_flight_ratio;
+                    let coin = (
+                        pay_coin.compute_object_reference(),
+                        pay_coin.owner,
+                        primary_gas.2.clone(),
+                    );
+                    let coin_type_tag = pay_coin.get_move_template_type()?;
                     let workloads = if !opts.disjoint_mode {
+                        let all_shared_counter_coin_configs = if shared_counter == 0 {
+                            None
+                        } else {
+                            let num_shared_counters =
+                                (max_ops as f32 * shared_counter_ratio) as u64;
+                            let shared_counter_init_coin_configs =
+                                SharedCounterWorkload::generate_coin_config_for_init(
+                                    num_shared_counters,
+                                );
+                            let shared_counter_payload_coin_configs =
+                                SharedCounterWorkload::generate_coin_config_for_payloads(max_ops);
+                            Some((
+                                shared_counter_init_coin_configs,
+                                shared_counter_payload_coin_configs,
+                            ))
+                        };
+                        let all_transfer_object_coin_configs = if transfer_object == 0 {
+                            None
+                        } else {
+                            Some(TransferObjectWorkload::generate_coin_config_for_payloads(
+                                max_ops,
+                                opts.num_transfer_accounts,
+                                in_flight_ratio * target_qps,
+                            ))
+                        };
+                        let (
+                            shared_counter_workload_init_gas_config,
+                            shared_counter_workload_payload_gas_config,
+                        ) = all_shared_counter_coin_configs.unwrap_or((vec![], vec![]));
+                        let (
+                            transfer_object_workload_tokens,
+                            transfer_object_workload_payload_gas_config,
+                        ) = all_transfer_object_coin_configs.unwrap_or((vec![], vec![]));
```

Comment:
Instead of unwrap_or here, why not just return empty vectors above when in the if branches?

Reply:
fixed